com/src/lib.rs

```diff
@@ -83,8 +83,8 @@ pub trait ProductionComInterface<T: IUnknown>: ComInterface {
 #[doc(hidden)]
 #[macro_export]
 macro_rules! vtable {
-    ($class:ident: $interface:ident, $offset:path) => {
-        <dyn $interface as $crate::ProductionComInterface<$class>>::vtable::<$offset>();
+    ($class:ident$( <$generics:tt>)?: $interface:ident, $offset:path) => {
+        <dyn $interface as $crate::ProductionComInterface<$class$( <$generics>)?>>::vtable::<$offset>();
```

Comment:
I'm not entirely sure if the macro parser is greedy or not, i.e. whether nested generics are parsed correctly. I couldn't get the macro to work without the explicit `<>`s though.